### PR TITLE
Core integrity api merge -> Develop

### PR DESCRIPTION
**Pull request by GD-ResilienX**
This is the combination of all the changes for the core integrity api epic.

### CODE REVIEW

**Review by AS-resilienx — APPROVED** [review]
Api looks good.

File: FraihmworkHealthAndIntegrityApi.yaml [hunk 1]
```diff
@@ -0,0 +1,1361 @@
...
+          type: string
+          description: The exact model string or number that describes this component
+        startupTime:
+          type: string
+          format: date-time
+          description: The time that this component last reached an operational state. If not provided, FRAIHMWORK will assume that the start time is last time it received a message with state = OPERATIONAL from this component. Follows format as specified by RFC 3339 section 5.6, which is a specific implementation of ISO 8601
+        parent:
+          type: string
+          format: uuid
+          description: This is an optional field that can be used to denote that this is a subsystem of a different component. If left blank, it is assumed that there is no parent for this component, otherwise, it is expected that the UUID provided corresponds to an already existing and registered component.
```
Comment (MS-ResilienX) [inline FraihmworkHealthAndIntegrityApi.yaml:1080]: For the sake of consistency, change "subsystem" to "subcomponent"

File: FraihmworkHealthAndIntegrityApi.yaml [hunk 2]
```diff
@@ -0,0 +1,1361 @@
...
+        - uuid
+        - name
+
+    State:
+      type: object
+      properties:
+        state:
+          type: string
+          description: "
+          This is a high level description of the component or subsystem state. <br> <br>
```
Comment (MS-ResilienX) [inline FraihmworkHealthAndIntegrityApi.yaml:1156]: subcomponent. Though, Subcomponents are components, maybe not go into topology in the State description?

File: FraihmworkHealthAndIntegrityApi.yaml [hunk 3]
```diff
@@ -0,0 +1,1361 @@
...
+
+        availability:
+          description: Same as TOTAL UPTIME % for now. Will update definition as we gain ability to inject modules for handling different types of components (like using additional properties defined for radars)
+          type: number
+          format: double
+        
+    Fault:
+      type: object
+      description: A fault describes a specific condition that is adversely affecting the reporting component or subsystem. FRAIHMWORK uses this information to alert maintainers of problems, or to integrate various faults to identify larger system of system problems. Faults are identified by FRAIHMWORK by their name, code and source. Therefore there are assumptions to how faults are reported that should be abided by. Namely - each unique fault should have the same name and code information. Updates to the same fault should be reusing these keys, but can allow for other properties to change over time. If a fault is no longer active, it is expected that the component provides an update notifying FRAIHMWORK that the fault has been cleared. Otherwise, the fault will automatically clear after a default time interval if there have been no updates.
```
Comment (MS-ResilienX) [inline FraihmworkHealthAndIntegrityApi.yaml:1247]: component or subcomponent

File: FraihmworkHealthAndIntegrityApi.yaml [hunk 4]
```diff
@@ -0,0 +1,1361 @@
...
+          type: string
+          description: A brief high level summary of the fault
+          
+        description:
+          type: string
+          description: A more detailed description of the fault with as much relevant information as possible. Implementors are encouraged to define a standard format for fault descriptions that works for their service
+        source:
+          type: string
+          format: uuid
+          description: The source UUID of the component that is causing the fault. Used when the client can detect failures with other components or subsystems. When not included, it is assumed that the source is the UUID of this reporting client
```
Comment (MS-ResilienX) [inline FraihmworkHealthAndIntegrityApi.yaml:1263]: component or subcomponents

**Review by MS-ResilienX — APPROVED** [review]
Recommended just a couple changes to not refer to systems in certain comments. Approving assuming you'll consider my suggestions.

**Pull request merged**